packages/flutter/test/widgets/autocomplete_test.dart

```diff
@@ -923,6 +987,89 @@ void main() {
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Scaffold(
+          body: RawAutocomplete<String>(
+            showOptionsViewOnUncompletedOptions: true,
+            optionsBuilder: (TextEditingValue textEditingValue) async {
+              final Iterable<String> options = kOptions.where((String option) {
+                return option.contains(textEditingValue.text.toLowerCase());
+              });
+              if (delay == null) {
+                return options;
+              }
+              isLoading = true;
+              final Iterable<String> optionsAfterDelay = await Future<Iterable<String>>.delayed(delay, () => options);
+              isLoading = false;
+              return optionsAfterDelay;
+            },
+            fieldViewBuilder: (BuildContext context, TextEditingController fieldTextEditingController, FocusNode fieldFocusNode, VoidCallback onFieldSubmitted) {
+              focusNode = fieldFocusNode;
+              textEditingController = fieldTextEditingController;
+              return TextField(
+                key: fieldKey,
+                focusNode: focusNode,
+                controller: textEditingController,
+              );
+            },
+            optionsViewBuilder: (BuildContext context, AutocompleteOnSelected<String> onSelected, Iterable<String> options) {
+              if (isLoading){
+                return Container(key: optionsLoadingKey);
+              }
+              lastOptions = options;
+              return Container(key: optionsKey);
+            },
+          ),
+        ),
+      )
+    );
+
+    // The field is always rendered, but the options are not unless needed.
+    expect(find.byKey(fieldKey), findsOneWidget);
+    expect(find.byKey(optionsKey), findsNothing);
+    expect(find.byKey(optionsLoadingKey), findsNothing);
+
+    // Enter text to build the options with delay.
+    focusNode.requestFocus();
+    delay = const Duration(milliseconds: 500);
+    await tester.enterText(find.byKey(fieldKey), 'go');
+    await tester.pump();
+
+    // The options have not yet been built.
+    //
+    // However, the options view is now displayed in the interval between
+    // the field being focused and the options being built since
+    // showOptionsViewOnUncompletedOptions is set to true.
+    expect(find.byKey(optionsKey), findsNothing);
+    expect(find.byKey(optionsLoadingKey), findsOneWidget);
+    expect(lastOptions, isNull);
+
+    // Await asynchronous options builder.
+    await tester.pumpAndSettle(delay);
+    expect(find.byKey(optionsKey), findsOneWidget);
+    expect(lastOptions, <String>['dingo', 'flamingo', 'goose']);
+
+    // Loading view has been replaced with options results.
+    expect(find.byKey(optionsLoadingKey), findsNothing);
+
+    // Enter text to rebuild the options without delay.
+    delay = null;
+    await tester.enterText(find.byKey(fieldKey), 'ngo');
+    await tester.pump();
+    expect(lastOptions, <String>['dingo', 'flamingo']);
```

Comment:
Nit: Might as well expect that optionsKey is visible and not optionsLoadingKey?

Reply:
Added `expect`s for those.